stablehlo/dialect/StablehloOps.td

```diff
@@ -2340,13 +2340,19 @@ def StableHLO_CustomCallOp: StableHLO_Op<"custom_call",
     Encapsulates an implementation-defined operation `call_target_name` that
     takes `inputs` and `called_computations` and produces `results`.
 
+    Depending on the API version there are two ways to pass extra bits of static
+    information to the external function:
+    1. Use `API_VERSION_TYPED_FFI` which allows passing a dictionary attribute.
+    2. Use a previous API version with a StringAttr to encode backend config.
+
     See:
     https://github.com/openxla/stablehlo/blob/main/docs/spec.md#custom_call
 
     Example:
     ```mlir
     %results = stablehlo.custom_call @foo(%input0) {
-      backend_config = "bar",
+      backend_config = {bar = 42 : i32},
+      api_version = 4 : i32,
```

Comment:
Can you also remove the comment from the spec.md file mainly for the consistency with the ODS and spec? As we are not providing any explicit specification of API version, it does not matter, from the exemplification POV, which API version we are using.

``` 
 backend_config = {bar = 42 : i32},
  // api_version 4 is to express backend_config as a dictionary attribute
  api_version = 4 : i32,
```